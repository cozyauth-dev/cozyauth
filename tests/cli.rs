use std::fs;

use assert_cmd::Command;
use insta::{assert_debug_snapshot, glob};
use serde::Deserialize;

#[derive(Deserialize, Debug)]
struct TestCmd {
    cmd: String,
    args: Option<Vec<String>>,
}

#[test]
fn test_snapshots() {
    glob!("cli/inputs/*.toml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let test_cmd: TestCmd = toml::from_str(&input).unwrap();
        let assert = Command::cargo_bin(test_cmd.cmd)
            .unwrap()
            .args(test_cmd.args.unwrap_or_default().as_slice())
            .assert();
        assert_debug_snapshot!(assert.get_output());
    });
}
