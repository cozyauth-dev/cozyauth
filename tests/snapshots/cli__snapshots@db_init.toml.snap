---
source: tests/cli.rs
expression: assert.get_output()
input_file: tests/cli/inputs/db_init.toml
---
Output {
    status: ExitStatus(
        unix_wait_status(
            0,
        ),
    ),
    stdout: "Init Postgres\n",
    stderr: "",
}
